molr-mole-core/src/main/java/io/molr/mole/core/tree/AbstractJavaMole.java

```diff
@@ -49,13 +49,29 @@ protected AbstractJavaMole(Set<Mission> availableMissions) {
     @Override
     public Mono<MissionHandle> instantiate(Mission mission, Map<String, Object> params) {
         return supplyAsync(() -> {
+            // TODO we should discuss where to put parameter validation
+            validateParameterValues(mission, params);
             MissionHandle handle = handleFactory.createHandle();
             executors.put(handle, executorFor(mission, params));
             instances.add(new MissionInstance(handle, mission));
             publishState();
             return handle;
         });
     }
+
+    private void validateParameterValues(Mission mission, Map<String, Object> params) {
+        MissionParameterDescription parameterDescription = missionParameterDescriptionOf(mission);
+        parameterDescription.parameters().forEach(parameter -> {
+            String parameterName = parameter.placeholder().name();
+            Object parameterValue = params.get(parameterName);
+            if(parameter.allowedValues() != null && !parameter.allowedValues().isEmpty()) {
```

Comment:
Potentially, we should document somewhere that an empty allowedValues() means that "everything is allowed"?

... Not sure if this is the ideal solution ... An alternative could be an additional boolean flag like 'restricted' and the set as 'options' then one could even have non-restricted parameters with options .... However, for the moment good as it is ;-)